spec.bs

```diff
@@ -2660,6 +2696,32 @@ CORP violation report=] algorithm, as leaving it unfenced may cause a privacy le
   [=automatic beacon data/once=] set to true.
 </div>
 
+<div algorithm=get-initiator-ancestor-root-origin>
+  To <dfn>get the initiator ancestor root origin</dfn> of a given {{Document}} |sourceDocument|:
+
+  1. Let |navigable| be |sourceDocument|'s [=node navigable=].
+
+  1. While |navigable| is not null and |navigable| is not a [=fenced navigable container/fenced
+     navigable=]:
+  
+    1. Set |navigable| to |navigable|'s [=navigable/parent=].
+
+  1. If |navigable| is null, return null.
+
+  1. Otherwise:
+
+    1. Set |navigable| to |navigable|'s [=navigable/unfenced parent=].
+
+    1. While |navigable| is not null and |navigable| is not a [=fenced navigable container/fenced
+      navigable=]:
+
+      1. Set |navigable| to |navigable|'s [=navigable/parent=].
+
+    1. If |navigable| is null, return null.
+
+       Otherwise, return |navigable|'s associated {{Document}}'s [=Document/origin=].
```

Comment:
Link between navigable -> Document is unclear, since what you linked to `{{Document}}` is not a member of navigable.